### PR TITLE
Add generic logic for filtering through enums

### PR DESCRIPTION
**Pull request by jamesfneyer**
I didn't provide any unit tests for this, as I wasn't sure if it was possible to have filtering if there wasn't a django model to do so with.

### CODE REVIEW

**Comment by ajhodges** [comment]
Check this out: https://github.com/dcramer/mock-django

File: src/shipchain_common/filters.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from rest_framework.exceptions import ValidationError
+
+
+def filter_enum(queryset, field, value):
+    enum = getattr(queryset.model, field).field.enum
```
[the remark this answers is not in the document]
Comment (jamesfneyer) [reply src/shipchain_common/filters.py]: I was testing against the later as that's the data that was coming through in actual use. I can test it in that way too though.

[the remark this answers is not in the document]
Comment (ajhodges) [reply src/shipchain_common/filters.py]: I'm not talking about test data, I'm talking about the parameters of this function.

File: .gitignore [hunk 2]
```diff
@@ -59,6 +59,7 @@ coverage.xml
 *.log
 local_settings.py
 db.sqlite3
+test_enumobject
```
Comment (ajhodges) [inline .gitignore]: What's this for?

Comment (jamesfneyer) [reply .gitignore]: its an outdated thing from the previous way I was mocking django, don't need to worry about it now!

**Pull request merged**